### PR TITLE
Resolve #347 -- Game is now pausable

### LINKED ISSUE #347: Make game pauseable
Make /pause and /unpause functional

### PR DESCRIPTION
**Pull request by FurkanS1821**
/pause and /unpause commands fully functional, if 30 minutes have past, the game will auto resume itself.

Refs #347

### CODE REVIEW

**Comment by molenzwiebel** [comment]
The pausing logic should really be moved somewhere else, because currently it is way too easy to accidentally forget the statement and thus having packets that get handled even if the game is paused.

**Comment by FurkanS1821** [comment]
@molenzwiebel Packets should be handled, or /unpause wouldn't work

**Comment by molenzwiebel** [comment]
Well yeah, but it'd still be a cleaner solution to move the unpause packet check to a higher level instead of checking for a paused game in every single packet handler. It's currently way too easy forget the check.

**Review by molenzwiebel — CHANGES_REQUESTED** [review]
See the comments.

**Review by MythicManiac — CHANGES_REQUESTED** [review]
Agreed with @molenzwiebel

**Review by molenzwiebel — APPROVED** [review]
Until we have #369 this is the best we can do with regards to checking if a packet needs to be paused. Seems good 👍

File: GameServerLib/Logic/Packets/PacketHandlers/Handlers/HandleStartGame.cs [hunk 1]
```diff
@@ -56,6 +56,11 @@ public bool HandlePacket(Peer peer, byte[] data)
                              * of updating a champion's level, but it works */
                             var levelUpPacket = new LevelUp(player.Item2.Champion);
                             _game.PacketHandlerManager.sendPacket(peer, levelUpPacket, Channel.CHL_S2C);
+                            if (_game.IsPaused)
```
Comment (MythicManiac) [inline GameServerLib/Logic/Packets/PacketHandlers/Handlers/HandleStartGame.cs:59]: Why is the addition here necessary? Isn't the packet sent in any case further below?

Comment (FurkanS1821) [reply GameServerLib/Logic/Packets/PacketHandlers/Handlers/HandleStartGame.cs:59]: This is sent to the reconnected clients

**Comment by FurkanS1821** [comment]
Closing this because it's not pointing to `indev`

**Pull request closed without merging**